src/fetcher.rs

```diff
@@ -64,7 +64,7 @@ impl TokenFetcher {
     /// currently stored token's `expires_in` field and the configured
     /// `refresh_buffer`. If it is, return the stored token. If not,
     /// fetch a new token, store it, and return the new token.
-    pub async fn fetch_token(&self) -> Result<Token> {
+    pub async fn fetch_token(&mut self) -> Result<Token> {
```

Comment:
Wondering about the impact of this change..

Currently `TokenFetcher` is `Send` and `Sync`, so it can be shared across threads to create tokens. This doesn't make much sense anymore, if `fetch_token()` requires mutable access to self, because we'd need a `Mutex` anyway to use it across threads.
As a library user if I wrap `TokenFetcher` in a `Mutex` it means the mutex will be locked for the full duration of the async call `fetch_token()`, which includes the API round-trip.

What do you think about using interior mutability, i.e. by wrapping the Jwt in an `Arc<Mutex<T>>` on `TokenFetcher`?

Here's a draft if you think this makes sense: https://github.com/durch/rust-goauth/pull/19

Reply:
Great suggestion, I wasn't really happy about interior mutability, thanks for jumping on it